pandas/core/indexes/range.py

```diff
@@ -315,6 +315,14 @@ def is_monotonic_decreasing(self):
     def has_duplicates(self):
         return False
 
+    def __contains__(self, key):
+        hash(key)
+        try:
+            key = com.ensure_python_int(key)
```

Comment:
A Python int is needed, else Python iterates over the range, making the operation very slow.